zcm/julia/ZCM.jl

```diff
@@ -0,0 +1,334 @@
+
+function encode end
+function decode end
+
+function good(zcm::Zcm)
+    return (zcm.zcm != C_NULL) && (errno(zcm) == 0);
+end
+export good;
+
+function errno(zcm::Zcm)
+    return ccall(("zcm_errno", "libzcm"), Cint, (Ptr{Native.Zcm},), zcm.zcm);
+end
+export errno;
+
+function strerror(zcm::Zcm)
+    val =  ccall(("zcm_strerror", "libzcm"), Cstring, (Ptr{Native.Zcm},), zcm.zcm);
+    if (val == C_NULL)
+        return "unable to get strerror";
+    else
+        return unsafe_string(val);
+    end
+end
+export strerror;
+
+# TODO: get this into docs:
+# Note to self: handler could be either a function or a functor, so long as it has
+#               handler(rbuf::RecvBuf, channel::String, usr) defined
+function subscribe(zcm::Zcm, channel::AbstractString, handler, usr)
+
+    handler_jl = __zcm_handler(handler, usr);
+    handler_c  = cfunction(__zcm_handler_wrapper, Void,
+                           (Ptr{Native.RecvBuf}, Cstring, Ptr{Void},));
+    uv_wrapper = ccall(("uv_zcm_msg_handler_create", "libzcmjulia"), Ptr{Native.UvSub},
+                       (Ptr{Void}, Any,), handler_c, handler_jl);
+    uv_handler = cglobal(("uv_zcm_msg_handler_trigger", "libzcmjulia"));
+
+    return Sub(ccall(("zcm_subscribe", "libzcm"), Ptr{Native.Sub},
+                     (Ptr{Native.Zcm}, Cstring, Ptr{Void}, Ptr{Native.UvSub},),
+                     zcm.zcm, channel, uv_handler, uv_wrapper),
+               handler_jl, handler_c, uv_handler, uv_wrapper);
+end
+export subscribe;
+
+# RRR: might be a way to make this template a bit prettier
+# TODO: get this into docs:
+# Note to self: handler could be either a function or a functor, so long as it has
+#               handler(rbuf::RecvBuf, channel::String, msg::MsgType, usr) defined
+function subscribe(zcm::Zcm, channel::AbstractString, MsgType::DataType, handler, usr)
+    return subscribe(zcm, channel,
+                     function (rbuf::RecvBuf, channel::AbstractString, usr)
+                         msg = decode(MsgType, rbuf.data)
+                         handler(rbuf, channel, msg, usr);
+                     end, usr);
+end
+export subscribe;
+
+function unsubscribe(zcm::Zcm, sub::Sub)
+    ret = ccall(("zcm_unsubscribe", "libzcm"), Cint,
+                 (Ptr{Native.Zcm}, Ptr{Native.Sub}), zcm.zcm, sub.sub);
+    ccall(("uv_zcm_msg_handler_destroy", "libzcmjulia"), Void,
+          (Ptr{Native.UvSub},), sub.uv_wrapper);
+    return ret;
+end
+export unsubscribe;
+
+function publish(zcm::Zcm, channel::AbstractString, data::Array{UInt8}, datalen::UInt32)
+    return ccall(("zcm_publish", "libzcm"), Cint,
+                 (Ptr{Native.Zcm}, Cstring, Ptr{Void}, UInt32),
+                 zcm.zcm, channel, data, datalen);
+end
+export publish;
+
+# TODO: force msg to be derived from our zcm msg basetype
+function publish(zcm::Zcm, channel::AbstractString, msg)
+    data = encode(msg)
+    return publish(zcm, channel, data, UInt32(length(data)));
+end
+export publish;
+
+function flush(zcm::Zcm)
```

Comment:
There is a `Base.flush` that you probably should be extending here (i.e. replace `flush` with `Base.flush`).